pyanaconda/network.py

```diff
@@ -230,8 +230,13 @@ def copy_resolv_conf_to_root(root="/"):
     dst_dir = os.path.dirname(dst)
     if not os.path.isdir(dst_dir):
         util.mkdirChain(dst_dir)
-    shutil.copyfile(src, dst)
 
+    try:
+        target = os.readlink(src)
+    except OSError:
+        shutil.copyfile(src, dst)
+    else:
+        os.symlink(target, dst)  # symlink() takes symlink path as second argument
```

Comment:
I wonder why the `else` clause is needed at all? Is it because you want to conditionalize only the errors from `readlink`? In that case, why not use a plain if-else with `os.path.islink`?
```
if os.path.islink(src):
    target = os.readlink(src)
    os.symlink(target, dst)  # symlink() takes symlink path as second argument
else:
    shutil.copyfile(src, dst)
```

Reply:
That version does more syscalls and suffers from a TOCTOU condition.
But a better fix is to use `follow_symlinks=True` in any case. I didn't know shutil.copyfile() supports that.

Anywa, IIUC,  #3230 obsoletes this PR.

Reply:
Oh, I didn't consider that... thanks!

Reply:
I updated the patch to use `follow_symlinks`, just in case anyone ever looks at this again. But let's close the PR.